From feba7dec50b5e407444a1079e9298c97d4b8fa73 Mon Sep 17 00:00:00 2001
From: Arindam Nath <arindam.nath@amd.com>
Date: Mon, 26 Sep 2016 13:19:18 +0530
Subject: [PATCH 0016/1722] drm/amd/dal: set cursor position only when target
 is present

Scenario: Connect a monitor to CZ based platform using
DP-DP connection. Boot into X server without any display
manager running (preferrably using xinit). Wait for the
system to enter idle state and monitor to go blank. Move
the mouse to wake the system from idle state. Notice that
the below messages appear in dmesg output.

[drm:dc_target_set_cursor_position] *ERROR* DC: dc_target is NULL!
[drm:dm_crtc_cursor_move] *ERROR* DC failed to set cursor position

This patch addresses this issue by making sure we only
set the cursor position when a real target is present.

Reported-by: Chandu-babu Namburu <chandu@amd.com>
Signed-off-by: Arindam Nath <arindam.nath@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
index ab8bc4c..f72116f 100644
--- a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
+++ b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
@@ -306,11 +306,13 @@ static int dm_crtc_cursor_move(struct drm_crtc *crtc,
 	position.x_hotspot = xorigin;
 	position.y_hotspot = yorigin;
 
-	if (!dc_target_set_cursor_position(
-				amdgpu_crtc->target,
-				&position)) {
-		DRM_ERROR("DC failed to set cursor position\n");
-		return -EINVAL;
+	if (amdgpu_crtc->target) {
+		if (!dc_target_set_cursor_position(
+					amdgpu_crtc->target,
+					&position)) {
+			DRM_ERROR("DC failed to set cursor position\n");
+			return -EINVAL;
+		}
 	}
 
 	return 0;
-- 
2.7.4

